src/App.tsx

```diff
@@ -1,9 +1,233 @@
+/* eslint-disable max-len */
 import './App.scss';
+import {
+  Routes, Route, Navigate,
+} from 'react-router-dom';
+import { useState, useEffect } from 'react';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header/Header';
+import {
+  ProductsListWithSlider,
+} from './components/Pages/Home/ProductsListWithSlider/ProductsListWithSlider';
+import {
+  ProductsSlider,
+} from './components/Pages/Home/ProductsSlider/ProductsSlider';
+import {
+  ShopByCategory,
+} from './components/Pages/Home/ShopByCategory/ShopByCategory';
+import { CartPage } from './components/Pages/Cart/CartPage';
+import { FavouritesPage } from './components/Pages/Favourites/FavouritesPage';
+import {
+  ProductsCardPage,
+} from
+  './components/Pages/Categories/ProductPagePage/ProductCardList/ProductsCardPage/ProductsCardPage';
+import {
+  ProductDetailsPage,
+} from
+  './components/Pages/Categories/ProductPagePage/ProductCardList/ProductDetailsPage/ProductDetailsPage';
+import { NotFoundPage } from './components/NotFoundPage/NotFoundPage';
+import { Product } from './types/types';
 
-const App = () => (
-  <div className="App">
-    <h1>React Phone Catalog</h1>
-  </div>
-);
+const App = () => {
+  const [IPhones, setIPhones] = useState([]);
+  const [shuffeledIPhones, setShuffeledIPhones] = useState([]);
+  const [visibleIPhones, setVisibleIPhones] = useState<Product[]>(IPhones);
+  const [hotPriceProducts, setHotPriceProducts] = useState([]);
+  const [brandNewProducts, setBrandNewProducts] = useState([]);
+  const [searchInput, setSearchInput] = useState('');
+
+  const getHotPriceProducts = () => {
+    setHotPriceProducts(
+      IPhones.filter((item: Product) => item.price < item.fullPrice)
+        .sort((a: Product, b: Product) => a.price - b.price),
+    );
+  };
+
+  const getBrandNewProducts = () => {
+    setBrandNewProducts(
+      IPhones.filter((item: Product) => item.price < item.fullPrice)
+        .sort((a: Product, b: Product) => b.price - a.price),
+    );
+  };
+
+  const getSuggestedProducts = () => {
+    setShuffeledIPhones(IPhones.sort(() => 0.5 - Math.random()));
+  };
+
+  useEffect(() => {
+    getBrandNewProducts();
+    getHotPriceProducts();
+    getSuggestedProducts();
+  }, [IPhones]);
+
+  const getIPhones = async () => {
+    try {
+      const response = await fetch(
+        'new/products.json',
+        {
+          method: 'GET',
+        },
+      );
+
+      if (response.status === 200) {
+        const result = await response.json();
+
+        setIPhones(result);
+        setVisibleIPhones(result);
+      }
+    } catch (err) {
+      throw new Error('Error');
+    }
+  };
+
+  useEffect(() => {
+    getIPhones();
+  }, []);
+
+  return (
+    <>
+      <Header
+        setVisibleIPhones={setVisibleIPhones}
+        IPhones={IPhones}
+        searchInput={searchInput}
+        setSearchInput={setSearchInput}
+      />
+      <Routes>
```

Comment:
all routes should be configured in separated file. You can make it as a component